src/Components/VideoCard/videoCard.js

```diff
@@ -1,34 +1,109 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
+import { Link, useLocation } from "react-router-dom";
+import { useHistory } from "../../Context/historyContext";
+import { useWatchLater } from "../../Context/watchLaterContext";
 
 const VideoCard = ({ video }) => {
   const [dropdown, setDropdown] = useState(false);
+
+  const location = useLocation();
   const trimHeading = (word, n) => {
```

Comment:
You Can create a utility export for such functions and can reduce the code repetition by importing from single place;
` export const trimHeading = (word, n) => {
    if (word.length > n) {
      return word.substring(0, n - 3) + "...";
    }
    return word;
  };`
  
  and Import it here!